internal/mode/static/state/graph/graph.go

```diff
@@ -183,6 +185,15 @@ func BuildGraph(
 
 	npCfg := getNginxProxy(state.NginxProxies, processedGwClasses.Winner)
 	gc := buildGatewayClass(processedGwClasses.Winner, npCfg, state.CRDMetadata, validators.GenericValidator)
+	if gc != nil && npCfg != nil {
+		for _, cond := range gc.Conditions {
```

Comment:
GC can still be valid if the parametersRef is not.

Reply:
Just because npCfg is not nil, does not mean it's valid. The validation happens when we build the GC, because the conditions live on the GC. So that's why we need to check the conditions.